plugins/synaptics-rmi/fu-synaptics-rmi-device.c

```diff
@@ -0,0 +1,1070 @@
+		priv->package_rev = fu_common_read_uint16 (f01_tmp->data + 2, G_LITTLE_ENDIAN);
+	}
+	if (has_build_id_query) {
+		g_autoptr(GByteArray) f01_tmp = NULL;
+		guint8 buf32[4] = { 0x0 };
+		f01_tmp = fu_synaptics_rmi_device_read (self, prod_info_addr, BUILD_ID_BYTES, error);
+		if (f01_tmp == NULL) {
+			g_prefix_error (error, "failed to read build ID bytes: ");
+			return FALSE;
+		}
+		if (!fu_memcpy_safe (buf32, sizeof(buf32), 0x0,		/* dst */
+				     f01_tmp->data, f01_tmp->len, 0x0,	/* src */
+				     f01_tmp->len, error))
+			return FALSE;
+		priv->flash.build_id = fu_common_read_uint32 (buf32, G_LITTLE_ENDIAN);
+	}
+
+	/* set main composite version */
+	fw_ver = g_strdup_printf ("%u.%u.%u",
+				  f01_basic->data[2],
+				  f01_basic->data[3],
+				  priv->flash.build_id);
+	fu_device_set_version (device, fw_ver, FWUPD_VERSION_FORMAT_TRIPLET);
+
+	priv->f34 = fu_synaptics_rmi_device_get_function (self, 0x34, error);
+	if (priv->f34 == NULL)
+		return FALSE;
+	f34_ctrl = fu_synaptics_rmi_device_read (self, priv->f34->control_base, CONFIG_ID_BYTES, error);
+	if (f34_ctrl == NULL) {
+		g_prefix_error (error, "failed to read the config id: ");
+		return FALSE;
+	}
+	priv->config_id = fu_common_read_uint32 (f34_ctrl->data, G_LITTLE_ENDIAN);
+
+	/* set up vfuncs for each bootloader protocol version */
+	if (priv->f34->function_version == 0x0) {
+		klass_rmi->setup = fu_synaptics_rmi_v5_device_setup;
+		klass_device->detach = fu_synaptics_rmi_v5_device_detach;
+		klass_device->write_firmware = fu_synaptics_rmi_v5_device_write_firmware;
+	} else if (priv->f34->function_version == 0x1) {
+		klass_rmi->setup = fu_synaptics_rmi_v6_device_setup;
+		klass_device->detach = fu_synaptics_rmi_v5_device_detach;
+		klass_device->write_firmware = fu_synaptics_rmi_v5_device_write_firmware;
+	} else if (priv->f34->function_version == 0x2) {
+		klass_rmi->setup = fu_synaptics_rmi_v7_device_setup;
+		klass_device->detach = fu_synaptics_rmi_v7_device_detach;
+		klass_device->write_firmware = fu_synaptics_rmi_v7_device_write_firmware;
+	} else {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "f34 function version 0x%02x unsupported",
+			     priv->f34->function_version);
+		return FALSE;
+	}
+
+	/* get Function34_Query0,1 */
+	if (!klass_rmi->setup (self, error)) {
+		g_prefix_error (error, "failed to read f34 queries: ");
+		return FALSE;
+	}
+	bl_ver = g_strdup_printf ("%u.0", priv->flash.bootloader_id[1]);
+	fu_device_set_version_bootloader (device, bl_ver);
+
+	/* in bootloader mode */
+	f01_db = fu_synaptics_rmi_device_read (self, priv->f01->data_base, 0x1, error);
+	if (f01_db == NULL) {
+		g_prefix_error (error, "failed to read the f01 data base: ");
+		return FALSE;
+	}
+	if (f01_db->data[0] & 0x40) {
+		fu_device_add_flag (device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER);
+	} else {
+		fu_device_remove_flag (device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER);
+	}
```

Comment:
for single line statements no need for braces

Reply:
We have this kind of thing in quite a few places, e.g. "add this flash if this boolean is set, otherwise remove it" -- but I can't come up with a good method name e.g. `fu_device_sync_flag (device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER, boolval);` -- do you think that's a good idea?